File: T3/proyectos/T3_PasoDatos/app/src/main/java/com/borja/t3_pasodatos/SecondActivity.java
package com.borja.t3_pasodatos;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

public class SecondActivity extends AppCompatActivity {

    TextView textoRecuperar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        instancias();
        recuperarDatos();
    }

    private void recuperarDatos() {
        if (getIntent().getExtras() != null) {
            String palabraRecuperada = (String) getIntent().getExtras().get(MainActivity.TAG_1);
            textoRecuperar.setText(palabraRecuperada);
        }
    }

    private void instancias() {
        textoRecuperar = findViewById(R.id.textoRecuperado);
    }
}

File: Kotlin/T2/T2_Inicio_RET/app/src/main/java/com/develop/t2_inicio_ret/SecondActivity.kt
package com.develop.t2_inicio_ret

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.TextView

class SecondActivity : AppCompatActivity() {

    private lateinit var nombreRecuperado: String
    private var edadRecuperada: Int = 0;
    private lateinit var correoRecuperado: String;

    private lateinit var textoNombre: TextView;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)
        instancias();
        recuperarDatos();
        iniciarUI()

    }

    private fun instancias() {
        textoNombre = findViewById(R.id.texto_recepcion)
    }

    private fun iniciarUI() {
        Log.v("datos_recuperado", nombreRecuperado)
        Log.v("datos_recuperado", correoRecuperado)
        Log.v("datos_recuperado", edadRecuperada.toString())
        textoNombre.setText(nombreRecuperado+ resources.getString(R.string.saludo_completo))
    }

    private fun recuperarDatos() {
        nombreRecuperado = intent.extras?.getString("nombre","defecto")
            ?: "defecto null bundle";
        edadRecuperada = intent.extras?.getInt("edad",16) ?: 0;
        correoRecuperado = intent.extras?.getString("correo","correo@defecto.com")
            ?: "sin correo"
    }
}